src/utils/commands/execute.ts

```diff
@@ -40,6 +41,20 @@ export function runCommandInMainThread(options: CommandOptions): Promise<Command
   });
 }
 
+export async function runCommandStreamInMainThread(listener: (...args: any[]) => void, options: CommandOptions) {
+  ensureRendererThread();
+  log.info('sending command stream to main thread', options);
+  const commandEventEmitter = new events.EventEmitter();
+  commandEventEmitter.addListener(options.commandId, listener);
+
+  const cb = (_event: unknown, arg: any) => {
+    if (arg.commandId !== options.commandId) return;
+    commandEventEmitter.emit(options.commandId, arg);
+  };
+  ipcRenderer.on('command-stream-event', cb);
+  ipcRenderer.send('run-command-stream', options);
```

Comment:
We should probably generate a unique id in this function and pass it to the `run-command-stream` via options.
Then, when we receive `command-stream-event` we check if the ID matches with the one we sent.
Otherwise I'm thinking that multiple calls to `runCommandStreamInMainThread` in different places could receive wrong data.

Reply:
I think you're correct, resolving this comment